### PR TITLE
Specify cog shop service

### PR DESCRIPTION
**Pull request by eriklien**
## Description
Added optional argument `cog_shop_service: Literal["prod", "staging"]` to `PowerOpsClient.__init__` and `CogShopAPI.__init__`

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/power-ops-sdk/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [pyproject.toml](https://github.com/cognitedata/power-ops-sdk/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/powerops/client/shop/cogshop_api.py [hunk 1]
```diff
@@ -18,16 +19,27 @@
...
 
         cluster = urlparse(self._cdf.config.base_url).netloc.split(".", 1)[0]
 
-        environment = ".staging" if project == "power-ops-staging" else ""
+        if self.cog_shop_service == "prod":
+            environment = ""
+        elif self.cog_shop_service == "staging":
+            environment = ".staging"
+        else:
+            environment = ".staging" if project == "power-ops-staging" else ""
```
Comment (notrixbe) [inline cognite/powerops/client/shop/cogshop_api.py:42]: environment = ".staging" if project == "power-ops-staging" else "" 
is this still needed ?

File: cognite/powerops/_version.py [hunk 2]
```diff
@@ -1 +1 @@
-__version__ = "0.104.2"
```
Comment (notrixbe) [inline cognite/powerops/_version.py:1]: would you not bump a minor version ? 0.105?

File: cognite/powerops/client/shop/cogshop_api.py (hunk 1 above)
Comment (eriklien) [reply cognite/powerops/client/shop/cogshop_api.py:42]: I wanted to leave the current logic as is (if `cog_shop_service` is not specified, then I think it still makes sense to use "staging" for "power-ops-staging" and prod for other projects (?))

**Review by notrixbe — APPROVED** [review]
LGFM

**Pull request merged**